### PR TITLE
Screen background sound #318

### PR DESCRIPTION
**Pull request by osamaasifoak**
Hi @michaelcspeed, please take a look at these items in this PR.

- Background screen repository added
- Background screen view model added
- Background screen added
- Audio Player provider added for background sound
- Ignore the player screen

### CODE REVIEW

File: lib/components/headers/collapsible_header_component.dart [hunk 1]
```diff
@@ -112,9 +112,12 @@ class _CollapsibleHeaderComponentState
     return Stack(
       fit: StackFit.expand,
       children: [
-        Image.asset(
-          image,
-          fit: BoxFit.fill,
+        // Image.asset(
```
Comment (michaelcspeed) [inline lib/components/headers/collapsible_header_component.dart]: Can we remove this comment?

File: lib/constants/strings/shared_preference_constants.dart [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class SharedPreferenceConstants {
```
Comment (michaelcspeed) [inline lib/constants/strings/shared_preference_constants.dart:1]: nice

File: lib/routes/routes.dart [hunk 3]
```diff
@@ -9,7 +10,8 @@ import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
 
 import '../views/home/home_wrapper_widget.dart';
-import '../views/player/player2/player_view.dart';
+import '../views/player_old/player2/player_view.dart';
+// import '../views/player/player_view.dart';
```
Comment (michaelcspeed) [inline lib/routes/routes.dart]: Remove comment?

File: lib/routes/routes.dart [hunk 4]
```diff
@@ -34,6 +36,7 @@ final router = GoRouter(
   routes: [
     GoRoute(
         path: HomePath,
+        // builder: (context, state) => BackgroundSoundView(),
```
Comment (michaelcspeed) [inline lib/routes/routes.dart]: also here

File: lib/routes/routes.dart [hunk 5]
```diff
@@ -174,7 +177,12 @@ MaterialPage<void> getCollectionMaterialPage(GoRouterState state) {
 }
 
 MaterialPage<void> getPlayerMaterialPage(GoRouterState state) {
-  return MaterialPage(key: state.pageKey, child: PlayerWidget());
+  var session = state.extra as SessionModel;
+  return MaterialPage(key: state.pageKey, child: PlayerWidget()
+      // child: PlayerView(
+      //   sessionModel: session,
+      // ),
```
Comment (michaelcspeed) [inline lib/routes/routes.dart]: also here

File: lib/views/background_sound/components/custom_track_shape_component.dart [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import 'package:Medito/constants/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class CustomTrackShapeComponent extends RectangularSliderTrackShape {
```
Comment (michaelcspeed) [inline lib/views/background_sound/components/custom_track_shape_component.dart]: I think we have multiple TrackShapes in the project now, so maybe calling it something like "BackgroundSoundVolumeTrackShape" would be better?

File: lib/views/player/components/bottom_action_component.dart [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+            () => {},
+          ),
+          width8,
+          _bottomActionLabel(
+            context,
+            'DONWLOAD',
+            () => {},
+          ),
+          width8,
+          // if (_hasBGSound())
```
Comment (michaelcspeed) [inline lib/views/player/components/bottom_action_component.dart]: need this?

File: lib/views/player/components/bottom_action_component.dart [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+          _bottomActionLabel(
+            context,
+            'SOUND',
+            () => {},
+          ),
+          // if (_hasBGSound()) width8,
```
Comment (michaelcspeed) [inline lib/views/player/components/bottom_action_component.dart]: need this?

File: lib/views/player/components/bottom_action_component.dart [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import 'package:Medito/constants/constants.dart';
+import 'package:flutter/material.dart';
+
+class BottomActionComponent extends StatelessWidget {
+  const BottomActionComponent({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final _speedList = ['X1', 'X1.25', 'X1.5', 'X2', 'X0.6'];
+    var _currentSpeed = 'X1';
```
Comment (michaelcspeed) [inline lib/views/player/components/bottom_action_component.dart:9]: Actually would be great if we could save the user's chosen speed and auto select that next time they load this screen?

**Review by michaelcspeed — COMMENTED** [review]
I'm wondering about the use of the word "component" - it seems a bit generic to me. Would be better to use "Widget" or "Shape" or something more descriptive, unless I'm missing something?

Other than that, looks great so far. Thank you!

File: lib/routes/routes.dart (hunk 3 above)
Comment (osamaasifoak) [reply lib/routes/routes.dart]: Actually, I was switching between the screens that's why I didn't remove the comment but we don't need it anymore so, I will remove all the comments thanks.

File: lib/views/background_sound/components/custom_track_shape_component.dart (hunk 6 above)
Comment (osamaasifoak) [reply lib/views/background_sound/components/custom_track_shape_component.dart]: Yes, you're right

File: lib/views/player/components/bottom_action_component.dart (hunk 9 above)
Comment (osamaasifoak) [reply lib/views/player/components/bottom_action_component.dart:9]: All these changes have been made in the player screen PR.

File: lib/views/player/components/bottom_action_component.dart (hunk 8 above)
Comment (osamaasifoak) [reply lib/views/player/components/bottom_action_component.dart]: removed

**Comment by michaelcspeed** [comment]
What do you think about my comment about the term "component" above? @osamaasifoak 

Is this PR ready to merge? Please ping me on Slack if so because I'm still not getting Github notifications :/

**Comment by osamaasifoak** [comment]
I think you're right. A single component contains multiple widgets and is much more complex than a widget. I will make this change to improve downloading feature PR. Let me know if it works for you

**Pull request merged**